messaginginapp/src/main/java/io/customer/messaginginapp/gist/presentation/engine/EngineWebView.kt

```diff
@@ -99,6 +99,7 @@ internal class EngineWebView @JvmOverloads constructor(
                     // post message to webview with the configuration data so that the message can be rendered
                     val script = """
                     window.postMessage($jsonString, '*');
+                    window.parent.postMessage = function(message) {window.${EngineWebViewInterface.JAVASCRIPT_INTERFACE_NAME}.postMessage(JSON.stringify(message))}
```

Comment:
```suggestion
                    val script = """
    // Post the JSON message to the current frame's listeners
    // Ensures internal JavaScript communication via window.addEventListener('message') remains functional
    window.postMessage($jsonString, '*');
    
    // Override window.parent.postMessage to route messages to the native Android interface
    // This is necessary only for legacy message because WebView can only attach one native interface 
    // (via addJavascriptInterface),
    // and we have already added it as ${EngineWebViewInterface.JAVASCRIPT_INTERFACE_NAME}.
    window.parent.postMessage = function(message) {
        window.${EngineWebViewInterface.JAVASCRIPT_INTERFACE_NAME}.postMessage($jsonString);
    }
""".trim()
```